implementations/rust/ockam/ockam_command/src/service/list.rs

```diff
@@ -67,21 +68,31 @@ async fn run_impl(
         .terminal
         .progress_output(&output_messages, &is_finished);
 
-    let (services, _) = try_join!(send_req, progress_output)?;
-
-    let plain = opts.terminal.build_list(
-        &services.list,
-        &format!("Services on {}", node_name),
-        &format!("No services found on {}", node_name),
-    )?;
-    let json = serde_json::to_string_pretty(&services.list).into_diagnostic()?;
-    opts.terminal
-        .stdout()
-        .plain(plain)
-        .json(json)
-        .write_line()?;
-
-    Ok(())
+    let result = try_join!(send_req, progress_output);
```

Comment:
We have a small utilty function to detect whether a node is running or not. Right after line 45 you can use the following snippet:

```rust
let node_name = parse_node_name(&node_name)?;
if !opts.state.nodes.get(&node_name)?.is_running() {
    return Err(miette!("The node {node_name} is not running"));
}
```

That way you don't have to check for specific errors/error messages.

Reply:
Ah that's much nicer! Thanks I'll make those changes.